src/js/model/schedule.js

```diff
@@ -288,8 +288,17 @@ Schedule.prototype.setAllDayPeriod = function(start, end) {
 };
 
 Schedule.prototype.setTimePeriod = function(start, end) {
-    this.start = new TZDate(start || Date.now());
-    this.end = new TZDate(end || this.start);
+    if (start instanceof TZDate) {
+        this.start = start;
```

Comment:
This is double checked statement compared with `createDateWithUTCTime()`.  `start instanceof TZDate` should be handled in `createDateWithUTCTime()` function.